### DISCUSSION TITLE
about the secondary-widget, I failed to add the widget

### DISCUSSION THREAD

**Discussion by tgfydzs** [zepp-health/zeppos-forum#327] [Q&A]
about the secondary-widget, I did not successfully add the widget. The operation is: add the index under widget, and add ‘secondary-widget’ under module in app.json. I followed the tutorial exactly. What is the problem?

  "targets": {
    "default": {
      "module": {
        "page": {
          "pages": [
            "page/index"
          ]
        },
        "app-side": {
          "path": "app-side/index"
        },
        "setting": {
          "path": "setting/index"
        },
        "secondary-widget": {
          "widgets": [
            {
              "path": "secondary-widget/index",
              "icon": "icon.png",
              "name": "widget-demo",
              "preview": "icon.png",
              "runtime": {
                "type": "js"
              }
            }
          ]
        }
      },
      "platforms": [
        {
          "st": "r",
          "dw": 480
        },
        {
          "st": "s",
          "dw": 390
        },
        {
          "st": "b",
          "dw": 194
        }
      ]
    }
  },

**Comment by tgfydzs** [comment]
Or is there a ready-made product for learning?

**Reply by MyoungXUE** [reply]
You can refer this project.

https://github.com/zepp-health/zeppos-samples/blob/main/application/2.0/calories/app.json

**Reply by tgfydzs** [reply]
I have referred to this official project, but there is no obvious difference after comparison, which can be seen in the app.json I provided.

**Comment by MyoungXUE** [comment]
What is your device? Not all devices support secondary screen applications. You can find the support on this document.

https://docs.zepp.com/docs/reference/related-resources/device-list/

**Reply by tgfydzs** [reply]
Balance

**Comment by tgfydzs** [comment]
I know, I need to add the widget in the settings of the simulator, but this is not the case on the device I have!

**Comment by MyoungXUE** [comment]
Has the problem been solved?
